src/Features/CSharp/Portable/Completion/KeywordRecommenders/AwaitKeywordRecommender.cs

```diff
@@ -51,6 +51,11 @@ protected override bool IsValidContext(int position, CSharpSyntaxContext context
                 return true;
             }
 
+            if (context.TargetToken.IsKind(SyntaxKind.UsingKeyword, SyntaxKind.ForEachKeyword))
```

Comment:
note: this will mean 'await' may be offered after 'using' at the top level.  That's probably not waht we want.  can you filter out that case and add a test.  (note: maybe we do want it, because of script... not sure).